pkg/crypto/wrapkey_opts.go

```diff
@@ -7,30 +7,35 @@ SPDX-License-Identifier: Apache-2.0
 package crypto
 
 type wrapKeyOpts struct {
-	senderKH interface{}
+	senderKey interface{}
 }
 
 // NewOpt creates a new empty wrap key option.
 // Not to be used directly. It's intended for implementations of Crypto interface
-// Use WithSenderKH() option function below instead.
+// Use WithSender() option function below instead.
 func NewOpt() *wrapKeyOpts { // nolint // unexported type doesn't need to be used outside of crypto package
 	return &wrapKeyOpts{}
 }
 
-// SenderKH gets the Sender key handle to be used for kew wrapping with a sender key (authcrypt).
-// Not to be used directly. It's intended for implementations of Crypto interface
-// Use WithSenderKH() option function below instead.
-func (pk *wrapKeyOpts) SenderKH() interface{} {
-	return pk.senderKH
+// SenderKey gets the Sender key to be used for key wrapping using a sender key (authcrypt).
+// Not to be used directly. It's intended for implementations of Crypto interface.
+// Use WithSender() option function below instead.
+func (pk *wrapKeyOpts) SenderKey() interface{} {
+	return pk.senderKey
 }
 
 // WrapKeyOpts are the crypto.Wrap key options.
 type WrapKeyOpts func(opts *wrapKeyOpts)
 
-// WithSenderKH option is for setting a sender key handle with crypto wrapping (eg: AuthCrypt). For Anoncrypt,
+// WithSender option is for setting a sender key with crypto wrapping (eg: AuthCrypt). For Anoncrypt,
 // this option must not be set.
-func WithSenderKH(senderKH interface{}) WrapKeyOpts {
+// Sender is a key used for ECDH-1PU key agreement for authenticating the sender.
+// senderkey can be of the following there types:
+//   - *keyset.Handle (requires private key handle for crypto.WrapKey())
+//   - *crypto.PublicKey (available for UnwrapKey() only)
+//   - *ecdsa.PublicKey (available for UnwrapKey() only)
+func WithSender(senderKey interface{}) WrapKeyOpts {
```

Comment:
Can we have 2 functions that are specific about input parameters instead of a single that is so generic?

Reply:
maybe at a later time, if this causes issues, for now I added the conditions where/how the option should be used.